plugins/woocommerce/src/Internal/StockNotifications/Notification.php

```diff
@@ -449,4 +449,117 @@ public function get_product_name() {
 
 		return $product->get_parent_id() ? $product->get_name() : $product->get_title();
 	}
+
+	/**
+	 * Check if the notification has valid verification data.
+	 *
+	 * This is used to determine if the notification can be verified via email link.
+	 *
+	 * @return bool
+	 */
+	public function has_valid_verification_data(): bool {
+		$action_key = $this->get_meta( 'email_link_action_key' );
+
+		if ( empty( $action_key) ) {
+			return false;
+		}
+
+		if ( ! str_contains( $action_key, ':' ) ) {
+			return false;
+		}
+
+		list( $requested_timestamp, $key ) = explode( ':', $action_key, 2 );
+		$expiration_duration = Config::get_unverified_deletion_days_threshold() * DAY_IN_SECONDS;
```

Comment:
I'm using the BIS setting to determine the threshold. Should we have a shorter threshold here? Maybe a day?